yt_dlp/extractor/maariv.py

```diff
@@ -0,0 +1,70 @@
+from .common import InfoExtractor
+import re
+
+
+class MaarivIE(InfoExtractor):
+    IE_NAME = 'maariv.co.il'
+    _VALID_URL = r'(https?://)?(www\.)?(m\.)?maariv\.co\.il(?:/[^/]+)*/Article-(?P<id>\d{7})'
+    _TESTS = [
+        {
+            'url': 'https://www.maariv.co.il/news/law/Article-1044008',
+            'info_dict': {
+                'id': '1044008',
+                'title': '.*',
+            },
+        }
+    ]
+
+    @staticmethod
+    def extract_resolution(url):
+        match = re.search(r'(\d{2,4}x\d{2,4})\.mp4$', url)
+        return match.group(1) if match else None
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        # Find the correct iframes
+        video_urls = re.findall(r'<iframe [^>]?poster="([^"]+)"[^>]+ src="([^"]+)"', webpage)
+        video_info_list = []
+
+        for thumbnail, src in video_urls:
```

Comment:
I would rename the variable and iterate over the JSON-LD objects instead:
```suggestion
        videos = []
        for video_obj in traverse_obj(self._yield_json_ld(webpage, video_id),
                                      lambda _, v: v['@type'] == 'VideoObject'):
```